crates/sozo/src/commands/options/mod.rs

```diff
@@ -1,10 +1,31 @@
-use scarb::core::Workspace;
+use scarb::core::{ManifestMetadata, Workspace};
 use toml::Value;
 
 pub mod account;
 pub mod starknet;
 pub mod world;
 
-pub(super) fn dojo_metadata_from_workspace(ws: &Workspace<'_>) -> Option<Value> {
-    ws.current_package().ok()?.manifest.metadata.tool_metadata.as_ref()?.get("dojo").cloned()
+pub(crate) fn dojo_metadata_from_workspace(ws: &Workspace<'_>) -> Option<DojoMetadata> {
+    Some(ws.current_package().ok()?.manifest.metadata.dojo_metadata())
+}
+
+pub(crate) struct DojoMetadata {
+    env: Option<Value>,
+}
+
+impl DojoMetadata {
+    pub fn env(&self) -> Option<Value> {
+        self.env.clone()
+    }
+}
+trait MetadataExt {
+    fn dojo_metadata(&self) -> DojoMetadata;
+}
+
+impl MetadataExt for ManifestMetadata {
+    fn dojo_metadata(&self) -> DojoMetadata {
+        let dojo_metadata = self.tool_metadata.as_ref().and_then(|e| e.get("dojo")).cloned();
+        let env_metadata = dojo_metadata.and_then(|inner| inner.get("env").cloned());
+        DojoMetadata { env: env_metadata }
```

Comment:
as mentioned, `toml::Value` provide a method `try_into` which can be used to deserialize it into type `T`

```suggestion
        self.tool_metadata
          .as_ref()
          .and_then(|e| e.get("dojo"))
          .cloned()
          .map(|v| v.try_into::<DojoMetadata>().unwrap_or_default())
          .unwrap_or_default()
```

we can just return the default value if `tool.dojo` can't be found